### PR TITLE
chore: disabling display location section 

### LINKED ISSUE #2104: Remove panic location from error messages
### Problem

image

The location text underneath is scary and unehlpful
Distracts the dev from the real error (

### Happy Case

Location should be where the error is coming from in my file!

### Alternatives Considered

_No response_

### Additional Context

_No response_

### Would you like to submit a PR for this Issue?

No

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by f01dab1e**
# Description

disabling display location section on release build

## Problem

Resolves #2104 

## Summary

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
Yeah, I'm getting a panic on compilation errors in Noir with this.

**Comment by f01dab1e** [comment]
as far as I understand, we are not panicking now, but returning an error:
```
[crates/nargo_cli/src/main.rs:16] start_cli() = Err(
    ReportedErrors(
        ReportedErrors {
            error_count: 1,
        },
    ),
)
```

the `HookBuilder` has the capability to handle both regular errors and panic situations, as demonstrated by the following code snippet:
```
eyre_hook.install()?;
panic_hook.install();
```

In the event of utilizing `RUST_BACKTRACE=1`, a backtrace is displayed; nevertheless, I am unable to identify any instances of panic during function calls.

File: crates/nargo_cli/src/main.rs [hunk 1]
```diff
@@ -7,9 +7,11 @@ const PANIC_MESSAGE: &str = "This is a bug. We may have already fixed this in ne
...
 fn main() -> eyre::Result<()> {
     // Register a panic hook to display more readable panic messages to end-users
-    let (panic_hook, _) =
-        HookBuilder::default().display_env_section(false).panic_section(PANIC_MESSAGE).into_hooks();
-    panic_hook.install();
+    HookBuilder::default()
+    .display_env_section(false)
+    .display_location_section(cfg!(debug_assertions))
+    .panic_section(PANIC_MESSAGE)
+    .install()?;
```
Comment (jfecher) [inline crates/nargo_cli/src/main.rs:14]: Does this work differently now that we're not calling `into_hooks` anymore? Does it mean we're installing to both the panic hook and the eyre hook now?

File: crates/nargo_cli/src/main.rs [hunk 2]
```diff
@@ -7,9 +7,11 @@ const PANIC_MESSAGE: &str = "This is a bug. We may have already fixed this in ne
 
 fn main() -> eyre::Result<()> {
     // Register a panic hook to display more readable panic messages to end-users
-    let (panic_hook, _) =
-        HookBuilder::default().display_env_section(false).panic_section(PANIC_MESSAGE).into_hooks();
-    panic_hook.install();
+    HookBuilder::default()
+    .display_env_section(false)
+    .display_location_section(cfg!(debug_assertions))
```
Comment (jfecher) [inline crates/nargo_cli/src/main.rs:12]:
```suggestion
    .display_location_section(false)
```
After experimenting with just always disabling this locally, I think we can go through with just never showing the Location portion. Actual panics will still print the stack trace so debugging is unaffected.

File: crates/nargo_cli/src/main.rs (hunk 1 above)
Comment (f01dab1e) [reply crates/nargo_cli/src/main.rs:14]: yeah

**Review by jfecher — APPROVED** [review]

**Review by TomAFrench — DISMISSED** [review]
If I compile a bad program such as

```
fn main(x: bool) {
    assert(foo);
}
```

I now get the output

```
$ nargo prove
error: cannot find `foo` in this scope 
  ┌─ /home/tom/Programming/noir/tmep/src/main.nr:2:12
  │
2 │     assert(foo);
  │            --- not found in this scope

warning: unused variable x
  ┌─ /home/tom/Programming/noir/tmep/src/main.nr:1:9
  │
1 │ fn main(x: bool) {
  │         - unused variable 

Error: 
   0: Aborting due to 1 previous error

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                                ⋮ 2 frames hidden ⋮                               
   3: nargo_cli::cli::start_cli::ha2a4506e189093d5
      at <unknown source file>:<unknown line>
   4: std::sys_common::backtrace::__rust_begin_short_backtrace::h0afdd9622daa4648
      at <unknown source file>:<unknown line>
   5: std::rt::lang_start::{{closure}}::h7a1a28d32aaaecbb
      at <unknown source file>:<unknown line>
   6: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h072eb4cd8da964ba
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/core/src/ops/function.rs:286
   7: std::panicking::try::do_call::h8eca204fe9266946
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/panicking.rs:483
   8: std::panicking::try::h12574e1b7b2cbacb
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/panicking.rs:447
   9: std::panic::catch_unwind::hf71522d4448329d6
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/panic.rs:137
  10: std::rt::lang_start_internal::{{closure}}::h65b66ac9bff580f8
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/rt.rs:148
  11: std::panicking::try::do_call::hfff61e33ca3db9f1
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/panicking.rs:483
  12: std::panicking::try::he48c8ecead279cad
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/panicking.rs:447
  13: std::panic::catch_unwind::hd510a26bfc950ccc
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/panic.rs:137
  14: std::rt::lang_start_internal::hc680b25eab888da9
      at /rustc/69f9c33d71c871fc16ac445211281c6e7a340943/library/std/src/rt.rs:148
  15: main<unknown>
      at <unknown source file>:<unknown line>
  16: __libc_start_call_main<unknown>
      at <unknown source file>:<unknown line>
  17: __libc_start_main_alias_1<unknown>
      at <unknown source file>:<unknown line>
  18: _start<unknown>
      at <unknown source file>:<unknown line>
```

I don't see how this is an improvement over the existing behaviour.

**Comment by f01dab1e** [comment]
this output will be only with "RUST_BACKTRACE=1"

(https://github.com/yaahc/color-eyre/blob/4a7b4d6988c6b0da5e04e29c9d6e10595b5dc302/src/config.rs#L1208-L1214)


```
> RUST_BACKTRACE=0 nargo check
warning: unused variable y
  ┌─ /home/work/subject/oliver/noir/h/src/main.nr:1:20
  │
1 │ fn main(x : Field, y : pub Field) {
  │                    - unused variable 

error: Expected type (), found type Field
  ┌─ /home/work/subject/oliver/noir/h/src/main.nr:1:4
  │
1 │ fn main(x : Field, y : pub Field) {
  │    ----


Error: 
   0: Aborting due to 1 previous error
   ```

**Comment by jfecher** [comment]
Is this happening because we're installing to the eyre-hook as well now? We should not print stack traces when issuing compiler errors under any circumstances, lets remove it. Many users may have `RUST_BACKTRACE=1` in their environment from working on other rust programs, they should not get a backtrace from the noir compiler as well.

**Comment by f01dab1e** [comment]
locations in this case should be disabled specifically for EyreHook, the workaround is to write your own report handler without printing the location

**Comment by jfecher** [comment]
Can we revisit this PR? The `Location` section is still showing up in error messages. Can we disable it entirely for debug and release builds without getting any RUST_BACKTRACE noise as well?

**Comment by kevaundray** [comment]
Possibly relevant is the disabling section in the readme: https://github.com/eyre-rs/color-eyre#disabling-tracing-support

**Comment by f01dab1e** [comment]
I tried through features, but it seems that because acvm-backend-barretenberg is enabled, it affects our crate

**Comment by jfecher** [comment]
Still getting some odd error formatting with this PR:
```
error: 
  ┌─ /.../noir/short/src/main.nr:3:12
  │
3 │     assert(false);
  │            ----- Failed constraint

Error: 
   0: Aborting due to 1 previous error
exit 1
```

**Comment by f01dab1e** [comment]
there is a miette, but there is the same problem: https://github.com/zkat/miette/issues/269

perhaps it would be easier to fork color_eyre?

**Comment by jfecher** [comment]
Looks like it may be printed by rust itself because we return a Result from main. Perhaps we should consider refactoring so that we do not do that?

If we print out the errors directly, then set the exit code with std::process::exit, would this still work on wasm builds? CC @TomAFrench

**Comment by TomAFrench** [comment]
If we're doing this in the `nargo_cli` crate then it's fine as that never gets compiled into wasm.

File: crates/nargo_cli/src/main.rs [hunk 3]
```diff
@@ -10,16 +10,13 @@
...
-    // Register a panic hook to display more readable panic messages to end-users
-    let (panic_hook, _) =
-        HookBuilder::default().display_env_section(false).panic_section(PANIC_MESSAGE).into_hooks();
-    panic_hook.install();
-
-    cli::start_cli()
+    if let Err(report) = cli::start_cli() {
+        eprintln!("{report}");
+        std::process::exit(1);
+    }
```
Comment (jfecher) [inline crates/nargo_cli/src/main.rs:21]: This looks much better!

File: crates/nargo_cli/src/panic.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        if let Ok(var) = std::env::var("RUST_BACKTRACE") {
+            if !var.is_empty() && var != "0" {
+                const HEX_WIDTH: usize = std::mem::size_of::<usize>() + 2;
+                // Padding for next lines after frame's address
+                const NEXT_SYMBOL_PADDING: usize = HEX_WIDTH + 6;
+                let mut backtrace = String::new();
+                let trace = Backtrace::new();
+                let frames = backtrace_ext::short_frames_strict(&trace).enumerate();
+                for (idx, (frame, sub_frames)) in frames {
+                    let ip = frame.ip();
```
Comment (jfecher) [inline crates/nargo_cli/src/panic.rs:39]: Can we switch back to color-eyre now that the miette experiment did not pan out? Looking at panic messages from both, I prefer the color-eyre formatting. It seems to require less code as well.

**Review by jfecher — COMMENTED** [review]
Thanks for the exploration here! I think this is a good approach we've settled on. Let's just revert back to color-eyre if possible.

File: crates/nargo_cli/src/panic.rs (hunk 4 above)
Comment (f01dab1e) [reply crates/nargo_cli/src/panic.rs]: we can only use color-eyre for panics. at least that way you don't have to fork color-eyre and eyre

Comment (jfecher) [reply crates/nargo_cli/src/panic.rs]: Why can't we remove miette entirely?

Comment (f01dab1e) [reply crates/nargo_cli/src/panic.rs]: it looks like I was wrong.

**Review by jfecher — APPROVED** [review]

**Comment by jfecher** [comment]
Pending re-review from @TomAFrench

**Comment by TomAFrench** [comment]
Will handle on Monday

**Review by kevaundray — APPROVED** [review]

**Pull request merged**